### PR TITLE
chore: Migrate .less styles to Emotion

### PR DESCRIPTION
**Pull request by kgabryje**
### SUMMARY
This PR migrates all .less files that target Dashboard, Explore and SqlLab. Generic stylesheets (for example with antd overrides or global LESS variables) are out of scope

### TESTING INSTRUCTIONS
Test all flows related to Dashboard and SqlLab. Compare with current master - there shouldn't be any meaningful changes, except for small spacing changes here and there (so that the spacings use grid unit from Superset's theme).
Explore shouldn't really be affected by this PR - I removed `explore/main.less` stylesheet, but it seems that it wasn't even used at all. However, check the Explore page just to be safe.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by michael-s-molina — COMMENTED** [review]
Thanks for the PR @kgabryje. I reviewed 20/61 files and will continue with the process but wanted to already publish some suggestions.

File: superset-frontend/src/dashboard/components/gridComponents/ChartHolder.tsx
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/gridComponents/ChartHolder.tsx:277]: Another way of writing this logic is to compile the CSS and apply it only if necessary:

```
const fullSizeStyle = css`
  && {
    position: fixed;
    z-index: 3000;
    left: 0;
    top: 0;
  }
`;

<div
   css={isFullSize ? fullSizeStyle : undefined}
>
```

File: superset-frontend/src/dashboard/components/gridComponents/Column.jsx
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/gridComponents/Column.jsx:76]: We need to translate the text here.

File: superset-frontend/src/dashboard/components/gridComponents/Divider.jsx
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/gridComponents/Divider.jsx:42]: Could you move `theme` to the top level and reuse it?

File: superset-frontend/src/dashboard/components/gridComponents/Header.jsx
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/gridComponents/Header.jsx:60]: Could you move `theme` to the top level and reuse it?

File: superset-frontend/src/dashboard/components/gridComponents/Row.jsx
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/gridComponents/Row.jsx:89]: Could you move theme to the top level and reuse it?

File: superset-frontend/src/dashboard/components/gridComponents/Row.jsx
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/gridComponents/Row.jsx:240]:
```suggestion
            <GridRow
              className={cx(
                rowItems.length === 0 && 'grid-row--empty',
```

File: superset-frontend/src/SqlLab/components/AceEditorWrapper/index.tsx
Comment (kgabryje) [reply superset-frontend/src/SqlLab/components/AceEditorWrapper/index.tsx:69]: I used those fonts because of the comment in line 68. In our theme monospace font we use Fira which is supposed to be bugged in Firefox

File: superset-frontend/src/dashboard/components/gridComponents/Column.jsx
Comment (kgabryje) [reply superset-frontend/src/dashboard/components/gridComponents/Column.jsx:76]: We need to move that text out of `::before` as simply doing `content: ${t('Empty column')};` doesn't work

File: superset-frontend/src/dashboard/components/gridComponents/Row.jsx
Comment (kgabryje) [reply superset-frontend/src/dashboard/components/gridComponents/Row.jsx:240]: We're using a `grid-row:after` selector in `DashboardBuilder`

File: superset-frontend/src/dashboard/components/resizable/ResizableContainer.jsx
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/resizable/ResizableContainer.jsx:93]:
```suggestion
      /* re-resizable sets an empty div to 100% width and height, which doesn't
      play well with many 100% height containers we need */
```

**Review by michael-s-molina — APPROVED** [review]
LGTM. Thank you for all the hard work and for addressing the comments @kgabryje. I would also suggest that you get QA's approval before merging the PR. We could schedule a team review to go through the application to see if we spot any differences as an additional precaution.

**Comment by kgabryje** [comment]
CC @ AnushaErrabelli for review

**Comment by AnushaErrabelli** [comment]
LGTM.

File: superset-frontend/src/SqlLab/components/AceEditorWrapper/index.tsx
Comment (rusackas) [reply superset-frontend/src/SqlLab/components/AceEditorWrapper/index.tsx:69]: We should probably make a follow-up PR that changes the global font, and still use the global one here, if Fira is problematic. Design folks might have opinions here, but I'll bet consistency is the most important thing.

File: superset-frontend/src/SqlLab/components/App/index.jsx
Comment (rusackas) [inline superset-frontend/src/SqlLab/components/App/index.jsx:93]: Seems like we might be able to get rid of this one, if I recall correctly.

File: superset-frontend/src/SqlLab/components/ResultSet/index.tsx
Comment (rusackas) [inline superset-frontend/src/SqlLab/components/ResultSet/index.tsx:452]: If this PR doesn't do it, we can probably do a sweep to get rid of these weird little utility classes soon.

File: superset-frontend/src/assets/stylesheets/superset.less
Comment (rusackas) [inline superset-frontend/src/assets/stylesheets/superset.less:316]: This probably got yanked in from another file, and i'm lost in the weeds, but just making sure we're not adding more of these utility classes if we don't need to ;)

File: superset-frontend/src/dashboard/components/DashboardBuilder/DashboardBuilder.tsx
Comment (rusackas) [inline superset-frontend/src/dashboard/components/DashboardBuilder/DashboardBuilder.tsx:101]: Is this what we'd want if we ever changed the scale of the grid unit? If it just needs to be 1px forever and always, so be it ;)

File: superset-frontend/src/dashboard/components/DashboardBuilder/DashboardBuilder.tsx
Comment (rusackas) [inline superset-frontend/src/dashboard/components/DashboardBuilder/DashboardBuilder.tsx:292]: At some point, we should bring back a z-index "registry" and centralize all these things values as vars.

File: superset-frontend/src/dashboard/components/dnd/DragDroppable.jsx
Comment (rusackas) [inline superset-frontend/src/dashboard/components/dnd/DragDroppable.jsx:82]: The theme contains opacity stops we should leverage wherever it makes sense. It includes 10% and 35% stops at the moment, so maybe we should try one of those here in a separate PR? There are probably random opacities all over we could audit for.

File: superset-frontend/src/dashboard/components/dnd/DragDroppable.jsx
Comment (rusackas) [inline superset-frontend/src/dashboard/components/dnd/DragDroppable.jsx:90]: Again, we should have a z-index registry, so we can think of all of Superset as layers in a consolidated way, do calculations like `theme.layers.navigation + 1`.

File: superset-frontend/src/dashboard/components/menu/WithPopoverMenu.tsx
Comment (rusackas) [inline superset-frontend/src/dashboard/components/menu/WithPopoverMenu.tsx:75]: Not sure if these should be in gridUnits, or if it's -42 for a specific reason.

File: superset-frontend/src/dashboard/styles.ts
Comment (rusackas) [inline superset-frontend/src/dashboard/styles.ts:21]: Hmm... I wonder if these are worth moving into (or along side of) the theme itself for global re-use

**Review by rusackas — APPROVED** [review]
This is awesome! I left a few nits/questions/follow-ups for future sweeping, but in general, this seems fantastic! Should we reset the baseline on visual regression tests and make sure this doesn't trigger any alarm bells?

**Comment by rusackas** [comment]
@kgabryje any reason not to merge this before the inevitable conflicts arise?

**Comment by kgabryje** [comment]
Good call @rusackas! I'll do a sweep through your comments in a followup PR

File: superset-frontend/src/dashboard/styles.ts
Comment (kgabryje) [reply superset-frontend/src/dashboard/styles.ts:21]: Previously it targeted only Dashboard, so I moved it here in order to reduce the impact and chances of regressions. You're right though that we might want to make it global

File: superset-frontend/src/dashboard/components/menu/WithPopoverMenu.tsx
Comment (kgabryje) [reply superset-frontend/src/dashboard/components/menu/WithPopoverMenu.tsx:96]: I tried to minimize refactoring of selectors to avoid regressions. However, I think our styles are due for an audit and refactoring/simplification

File: superset-frontend/src/dashboard/components/dnd/DragDroppable.jsx
Comment (kgabryje) [reply superset-frontend/src/dashboard/components/dnd/DragDroppable.jsx:90]: 👍

File: superset-frontend/src/dashboard/components/DashboardBuilder/DashboardBuilder.tsx
Comment (kgabryje) [reply superset-frontend/src/dashboard/components/DashboardBuilder/DashboardBuilder.tsx:292]: 👍

File: superset-frontend/src/dashboard/components/DashboardBuilder/DashboardBuilder.tsx
Comment (kgabryje) [reply superset-frontend/src/dashboard/components/DashboardBuilder/DashboardBuilder.tsx:101]: Good point 🤔 We should do a sweep through the styles and verify if some values should be hardcoded rather than be dependent on theme

File: superset-frontend/src/assets/stylesheets/superset.less
Comment (kgabryje) [reply superset-frontend/src/assets/stylesheets/superset.less:316]: Yeah, the "global" stylesheets were out of scope of this PR and that `.m-l-2` class looked like it belonged here.

File: superset-frontend/src/SqlLab/components/ResultSet/index.tsx
Comment (kgabryje) [reply superset-frontend/src/SqlLab/components/ResultSet/index.tsx:452]: Yup, out of scope of this PR

File: superset-frontend/src/SqlLab/components/App/index.jsx
Comment (kgabryje) [reply superset-frontend/src/SqlLab/components/App/index.jsx:93]: I'll check!

File: superset-frontend/src/dashboard/components/dnd/DragDroppable.jsx
Comment (kgabryje) [reply superset-frontend/src/dashboard/components/dnd/DragDroppable.jsx:82]: Ok, I'll see which fits better 👍

File: superset-frontend/src/dashboard/components/menu/WithPopoverMenu.tsx
Comment (kgabryje) [reply superset-frontend/src/dashboard/components/menu/WithPopoverMenu.tsx:57]: It's this border when component is focused in edit mode. Looks weird with 1px

image

File: superset-frontend/src/dashboard/components/menu/WithPopoverMenu.tsx
Comment (kgabryje) [reply superset-frontend/src/dashboard/components/menu/WithPopoverMenu.tsx:75]: Yes it should be in gridUnits. It's related to component's height (40px or 10gridUnits)

**Pull request merged**
